tests/unit/cli/admin/test_ceremony.py

```diff
@@ -735,6 +735,67 @@ def test_ceremony_option_bootstrap_upload(self, client, test_context):
             )
         ]
 
+    def test_ceremony_option_bootstrap_upload_no_auth(
+        self, client, test_context
+    ):
+        ceremony.bootstrap_status = pretend.call_recorder(
+            lambda *a: {"data": {"bootstrap": False}}
+        )
+        ceremony._load_bootstrap_payload = pretend.call_recorder(
+            lambda *a: {"k": "v"}
+        )
+        ceremony._send_bootstrap = pretend.call_recorder(
+            lambda *a: "fake_task_id"
+        )
+        ceremony.task_status = pretend.call_recorder(lambda *a: None)
+
+        test_context["settings"].AUTH = False
+
+        test_result = client.invoke(
+            ceremony.ceremony,
+            ["--bootstrap", "--upload", "--upload-server", "http://rstuf"],
+            input=None,
+            obj=test_context,
+        )
+
+        assert test_result.exit_code == 0, test_result.output
+        assert (
+            "Bootstrap completed using `payload.json`. 🔐 🎉"
+            in test_result.output
+        )
+        assert ceremony.bootstrap_status.calls == [
+            pretend.call(test_context["settings"])
+        ]
+        assert ceremony._load_bootstrap_payload.calls == [
+            pretend.call("payload.json")
+        ]
+        assert ceremony._send_bootstrap.calls == [
+            pretend.call(test_context["settings"], {"k": "v"})
+        ]
+        assert ceremony.task_status.calls == [
+            pretend.call(
+                "fake_task_id", test_context["settings"], "Bootstrap status: "
+            )
+        ]
+
+    def test_ceremony_option_bootstrap_upload_no_auth_missing_upload_server(
+        self, client, test_context
+    ):
+        test_context["settings"].AUTH = False
+
+        test_result = client.invoke(
+            ceremony.ceremony,
+            ["--bootstrap", "--upload"],
+            input=None,
+            obj=test_context,
+        )
+
+        assert test_result.exit_code == 1, test_result.output
+        assert (
+            "Requires '--upload-server' when using '--no-auth'"
```

Comment:
I think this test is not a regression test.
This error message is before the check for `upload_server` which you actually change.
I think in order to make a real regression test you need to add authorization and verify that the `settings.SERVER` will be equal to `--upload-server` which is not provided, so it will be `None` instead of being the real value stored inside `rstuf.ini`.

Reply:
Thank you for the observation!
I added the check in the last commit.

In case of regression, it raises this error
    
```shell
        # test regression https://github.com/vmware/repository-service-tuf-cli/pull/259  # noqa
>       assert test_context["settings"].SERVER is not None
E       assert None is not None
E        +  where None = <dynaconf.base.LazySettings object at 0x10e08f0a0>.SERVER

tests/unit/cli/admin/test_ceremony.py:739: AssertionError
```